pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -568,17 +666,21 @@ public void setDeletedSegmentsRetentionInDays(int retentionInDays) {
   }
 
   public int getTaskManagerFrequencyInSeconds() {
-    return getProperty(ControllerPeriodicTasksConf.TASK_MANAGER_FREQUENCY_IN_SECONDS,
-        ControllerPeriodicTasksConf.DEFAULT_TASK_MANAGER_FREQUENCY_IN_SECONDS);
+    return Optional.ofNullable(getProperty(ControllerPeriodicTasksConf.TASK_MANAGER_FREQUENCY_PERIOD))
```

Comment:
Does this work if value is set to "-1" in the new config (that is the default value).

Reply:
Handled this in the latest commit and added a UT to catch this. This wasn't handled earlier.

Reply:
I am sure I am missing something. I don't see the UT for this case?

Reply:
ah, got it.